test/functional/graders_controller_test.rb

```diff
@@ -349,7 +372,32 @@ class GradersControllerTest < AuthenticatedControllerTest
           assert @criterion3.tas.include? @ta3
         end
       end # flexible criteria
-    end # criteria csv upload
+
+      should 'gracefully handle malformed csv files' do
+        tempfile = fixture_file_upload('files/malformed.csv')
+        post_as @admin,
+                :csv_upload_grader_criteria_mapping,
+                assignment_id: @assignment.id,
+                grader_criteria_mapping: tempfile,
+                encoding: 'UTF-8'
+
+        assert_response :redirect
+        assert_equal flash[:error], I18n.t('csv.upload.malformed_csv')
+      end
+
+      should 'gracefully handle a non csv file with a csv extension' do
+        tempfile = fixture_file_upload('files/pdf_with_csv_extension.csv')
+        post_as @admin,
+                :csv_upload_grader_criteria_mapping,
+                assignment_id: @assignment.id,
+                grader_criteria_mapping: tempfile,
+                encoding: 'UTF-8'
+
+        assert_response :redirect
+        assert_equal flash[:error], I18n.t('csv.upload.non_text_file_with_csv_extension')
+      end
+
+      end # criteria csv upload
```

Comment:
`end` at 400, 6 is not aligned with `context 'doing a POST on :csv_upload_grader_criteria_mapping' do` at 231, 4